src/deltalake_ext.rs

```diff
@@ -335,32 +380,130 @@ fn delta_stats_from_file_metadata(file_metadata: &FileMetaData) -> Result<Stats,
             .filter_map(|g| g.column(i).statistics())
             .collect();
 
-        let (min, max) = stats_min_and_max(&statistics, column_descr.clone())?;
+        let column_path = column_descr.path();
+        let column_path_parts = column_path.parts();
+
+        let _ = apply_min_max_for_column(
+            statistics.as_slice(),
+            column_descr.clone(),
+            column_path_parts,
+            &mut min_values,
+            &mut max_values,
+        )?;
+    }
 
-        if let Some(min) = min {
-            let min = ColumnValueStat::Value(min);
-            min_values.insert(column_descr.name().to_string(), min);
-        }
+    Ok((min_values, max_values))
+}
 
-        if let Some(max) = max {
-            let max = ColumnValueStat::Value(max);
-            max_values.insert(column_descr.name().to_string(), max);
-        }
+fn apply_null_counts(array: &StructArray, null_counts: &mut HashMap<String, ColumnCountStat>) {
+    let fields = match array.data_type() {
+        DataType::Struct(fields) => fields,
+        _ => unreachable!(),
+    };
 
-        let null_count: u64 = statistics.iter().map(|s| s.null_count()).sum();
-        let null_count = ColumnCountStat::Value(null_count as i64);
-        null_counts.insert(column_descr.name().to_string(), null_count);
+    for i in 0..array.num_columns() {
```

Comment:
nit: this could be written as:

```rust
array.columns().iter().zip(fields).for_each(|(column , field)| { ... })
```

to simplify getting `column` and `field`

Reply:
:heart: this recommendation. fixed in latest commit.